src/audio/chain_dma.c

```diff
@@ -256,9 +256,12 @@ static enum task_state chain_task_run(void *data)
 				chain_get_transferred_data_size(link_read_pos,
 								host_read_pos,
 								buff_size);
+			size_t host_consume = MIN(transferred, host_avail_bytes);
```

Comment:
is it really happen that the host transferred more than what is available?

Reply:
This (always?) happens at the end of transfer. When host finished with streaming playback, host_avail_bytes is 0 or small value, but output HDA link still has a lot of older data to transfer. Without such check on test test_06_05_pass_through_chain_hda_codec_loopback I observe output stream is few times longer then input stream with last chunk of data repeated multiple times at the end of stream.
So ignoring host_avail_bytes transfer has difficulty to stop and continues to transfer last buffer contents.